### ISSUE TITLE
Multiple backdrop'd dialogs don't clean up properly

### ISSUE THREAD

**Issue by WrinklyNinja** [Polymer/core-overlay#37]
In Polymer 0.5.1, if I have more than one dialog with a backdrop open, at least one backdrop gets left behind when the dialogs are closed, and I often (but not always) get a `Current core-overlay is attempting to focus itself as next! (bug)` error message in the console, on line 506 of `core-overlay.html`.

Because the backdrop sits on top of everything, this makes it impossible to interact with the UI unless I go into the dev tools and remove the `<div class="core-overlay-backdrop">` element(s). If there are multiple such elements, one of them will also have the `core-opened` class, but not the others.

This is with non-layered overlays, I haven't tried it with layered overlays, so I don't know if that makes a difference.

**Comment by Ortham** [comment]
I looked into this, and found the following.

Scrim open/close/addition/removal is controlled by the `overlays` array. However, there is one scrim element per registered element (eg. `paper-dialog`, `foo-dialog`), whereas there is one `overlays` array for all elements.

If an element opens before another of a different type, then it sees no existing scrim, and attaches a new one. It sees that there are multiple backdrops, so doesn't toggle the new scrim open. The first element then closes, seeing that there is another backdrop open, so doesn't toggle the scrim.

The problem arises when the second element closes. It sees there are no backdrops, and tries to detach its scrim. However, the scrim never got attached, so it causes an error (`cannot read property 'removeChild' of null`), and the first element's scrim stays open.

Solutions:
1. Have one scrim for all elements, regardless of type.
2. When checking for backdrops, check for others of the same type, rather than just numbers.

I chose to implement the second solution, as I'd get Polymer errors when trying to set the scrims in the create or attach callbacks.
